pkg/backup/compaction_test.go

```diff
@@ -441,6 +449,13 @@ func TestBackupCompactionLocalityAware(t *testing.T) {
 					}},
 				},
 			},
+			ServerArgs: base.TestServerArgs{
```

Comment:
what additional test coverage does this add now that we have the unit test? just to check m.IsCompacted gets set? What if instead we ran a restore twice  with and without the setting set and asserted that the compacted backups were excluded/included?

Reply:
ah, i don't think you even need a testing knob for this (last resort). just count the number of URIs is the restore details.

Reply:
> just count the number of URIs in the restore details.

Oh yeah that works.

My intention behind adding this test coverage was to ensure that restore actually picked up those compacted backups. We had tests in place to ensure a compacted backup was created, but because the original incrementals were still there, we could have false positives where the restore and validation succeeded, but only because it restored off of the original backups.

Reply:
yeah, i think it would be a good idea to write this E2E test that validates the URI count